qiskit/algorithms/evolvers/classical_methods/scipy_imaginary_evolver.py

```diff
@@ -0,0 +1,143 @@
+from ..imaginary_evolver import ImaginaryEvolver
+from .scipy_evolver import SciPyEvolver
+
+
+class SciPyImaginaryEvolver(ImaginaryEvolver, SciPyEvolver):
+    r"""Classical Evolver for imaginary time evolution.
+
+    Evolves an initial state :math:`|\Psi\rangle` for an imaginary time :math:`\tau = it`
+    under a Hamiltonian  :math:`H`, as provided in the ``evolution_problem``.
+
+    For each timestep we use a taylor expansion of :math:`\exp(- \Delta \tau H)` to evolve the system.
+
+    """
+
+    def __init__(
+        self,
+        timesteps: Optional[int] = 100,
+    ):
+        """
+        Args:
+            timesteps: The number of timesteps in the simulation.
+
+        Raises:
+            ValueError: If `timesteps` is not a positive integer.
+        """
+        if isinstance(timesteps, int) and timesteps < 1:
+            raise ValueError("`timesteps` must be a positive integer.")
+
+        self.timesteps = timesteps
+
+    def evolve(self, evolution_problem: EvolutionProblem) -> EvolutionResult:
+        r"""Perform imaginary time evolution :math:`\exp(- \tau H)|\Psi\rangle`.
+
+        Args:
+            evolution_problem: The definition of the evolution problem.
+
+        Returns:
+            Evolution result which includes an evolved quantum state.
+
+        Raises:
+            ValueError: If a `NaN` is encountered in the final state of the system.
+            RuntimeError: If the Hamiltonian in the evolution problem is time dependent.
+            RuntimeError: If the state norm goes to 0 during time evolution
+        """
+
+        if evolution_problem.t_param is not None:
+            raise ValueError("Time dependent Hamiltonians are not currently supported.")
+
+        # Get the initial state, Hamiltonian and the auxiliary operators.
+        state, ev_operator, aux_ops = self._sparsify(evolution_problem=evolution_problem)
+
+        # Create empty arrays to store the time evolution of the auxiliary operators.
+        operators_number = (
+            0 if evolution_problem.aux_operators is None else len(evolution_problem.aux_operators)
+        )
+        ops_ev_mean = np.empty(shape=(operators_number, self.timesteps + 1), dtype=complex)
+
+        for ts in range(self.timesteps):
+            ops_ev_mean[:, ts] = self._evaluate_aux_ops(aux_ops, state)
+            state = ev_operator @ state
+
+            if np.nan in state:
+                raise RuntimeError(
+                    "An overflow has probably occured. Try increasing the amount of timesteps."
+                )
+            # We need to normalize first with respect to the max value to avoid having a norm
+            # with a value that overflows.
+            state = state / state.max()
+            # We then check that the state hasn't gone to zero.
+            state_norm = np.linalg.norm(state)
+            if state_norm == 0:
+                raise RuntimeError("The norm of the state went to 0.")
```

Comment:
Unlike the Nan error above there is no remediation suggestion here. Is there anything further we can say? Maybe we cannot evolve the problem as defined?